core/src/main/java/org/apache/druid/math/expr/ExprEval.java

```diff
@@ -1148,6 +1283,66 @@ public ExprType type()
       return ExprType.STRING_ARRAY;
     }
 
+    @Override
+    public boolean isNumericNull()
+    {
+      if (value != null && value.length == 1) {
+        return computeNumber(value[0]) == null;
```

Comment:
`computeNumber` seems expensive. Since the caller pattern is likely calling this and then one of `asX()` method below, should we cache the result?